204 and 376 add snow and conditions updates

CSDGC opened the pull request:
As per #204 and #376, snow has been added and Mission Conditions no longer resets the date.

darkChozo's cold breath particle spawner has also been added.

SamLex commented on f/missionConditions/f_setMissionConditions.sqf:
```diff
@@ -14,6 +14,12 @@ if (isServer) then {
+	// COLD BREATH
+	// Cold breath particle spawner for if it's snowing, or in low ambient temperatures
+	if (f_param_weather == 9 || f_param_weather == 10 || (ambientTemperature select 0) < 8) then {
+		remoteExec ["f_fnc_coldBreath",0,true];
```
```suggestion
		remoteExec ["f_fnc_coldBreath",-2,true];
```
Minor: Maybe exclude the server since this is a client-side effect?

NikkoJT replied:
Should be `[0,-2] select isDedicated` otherwise it will cause trouble in local-hosted MP

SamLex replied:
Good point. Ignore me, this isn't worth changing since the function has a `hasInterface` check anyway

SamLex commented on f/missionConditions/fn_ColdBreath.sqf:
```diff
@@ -0,0 +1,21 @@
+          velocityModelSpace _this vectorAdd [0, .15, 0], 1, 1.3, 1, .01, [.1,.22,.3,.35], [[1, 1, 1, 0.25],[1, 1, 1, 0]], [1], 1, 0, "", "", _this];
+    };
+  };
+} forEach allUnits;
```
```suggestion
} forEach f_var_men;
```
Minor: maybe the common F3 variable instead of `allUnits`? Likely functionally the same but more inline with other F3 code

SamLex replied:
Ignore this, apparently `f_var_men` is server-only (for some reason...)

SamLex commented on f/missionConditions/fn_ColdBreath.sqf:
```diff
@@ -0,0 +1,21 @@
+   _x spawn {
+     while {alive _this} do {
+        sleep (4*(1 - getFatigue _this) + random 1);
+        drop [["\A3\data_f\ParticleEffects\Universal\Universal", 16, 12, 8,1], "", "Billboard", 1,
```
Gonna just trust all this effect code :joy:

SamLex commented on f/missionConditions/fn_SetTime.sqf:
```diff
@@ -25,14 +25,10 @@ if ( _timeOfDay == 8 ) exitWith {};
+// GET DEFAULT VALUES
+// Grab the mission's current date and time.
+
+date params ["_year", "_month", "_day", "_hour", "_minute"];
```
Very neat! I like this

SamLex reviewed (approved):
Looks good to me! ~~A few minor comments but nothing blocking, feel free to ignore them~~

NikkoJT commented:
Is there a particular reason why coldBreath was changed from being a function to being execVM'd? Generally I'm in favour of more things being functions where possible as it's more efficient.

CSDGC commented:
No, I just only ever made it a function for easier remoteExec'ing, so when I removed one, I removed the other. It does make more sense to keep it as a function.
